text/0000-update-version-handling.md

```diff
@@ -0,0 +1,273 @@
+
+The `ToolSpec` change will make the format `tool[@version]` work; updating the commands to take a `Vec<ToolSpec>` will support passing multiple arguments.
+
+# Critique
+[critique]: #critique
+
+This introduces breaking changes—something which should not be done lightly! It does so under the assumption that fetching, installing, or pinning multiple tools simultaneously is a relatively common operation (relative to the frequency of running those operations at all). This assumption may be wrong.
+
+It also introduces a move away from users’ existing habits with `nvm` and `nodenv` and similar tools. This is thus an increase in the learning/switching costs for the tool.
+
+*Not* doing this is not likely to be a blocker for wider adoption. The “problem” identified here represents a papercut, not a major pain point.
+
+## Alternatives
+
+-   Support *both* invocation styles. This may *just* be technically feasible, allowing users to pass any of the following invocations successfully:
+
+    ```sh
+    notion pin create-react-app
+    notion pin create-react-app@3.0
+    notion pin create-react-app 3.0
+    notion pin create-react-app typescript
+    notion pin create-react-app@3.0 typescript
+    notion pin create-react-app typescript@3.4
+    notion pin create-react-app@3.0 typescript@3.4
+    ```
+
+    However, this introduces substantial additional complexity for the implementation, and ambiguity around certain invocations, e.g. for tagged versions/special names. For example, how should the following invocation be resolved?
+    
+    ```sh
+    notion pin create-react-app latest
+    ```
+
+    Does it refer to the latest version of `create-react-app`, or the default version of `create-react-app` and [the `latest` package](https://www.npmjs.com/package/latest)? For all non-SemVer-string parses, this will be impossible to resolve.
+
+    The proposed design here is well-proven with other tools which install more than simply a single tool-and-version pair, e.g. `npm`, `yarn`, and `cargo-edit`, and it does not fall prey to these problems of indeterminacy.
+
+-   Leave things as they are. This has the upside of matching other Node version managers, all of which use a close variant on our current syntax. However, none of those tools support installing anything but Node, so the comparison is less telling than it might seem.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+-   How do we handle errors when installing multiple things?
```

Comment:
I suggest we again match what npm and Yarn do, which appears to be to abort on the first error.